nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/ResponseHelper.java

```diff
@@ -79,11 +80,18 @@ public static Response blob(Blob blob, int httpStatus) {
         if (type == null || "???".equals(type)) {
             type = MediaType.APPLICATION_OCTET_STREAM;
         }
-        return Response.status(httpStatus)
+        Response response = Response.status(httpStatus)
                        .entity(blob)
                        .type(type)
                        .header("Content-Disposition", "attachment; filename=" + blob.getFilename())
                        .build();
+        // set Content-Type header with mimetype and charset
+        if (blob.getEncoding() != null) {
+            List<Object> contentType = new ArrayList<Object>();
+            contentType.add(type + "; charset=" + blob.getEncoding());
+            response.getMetadata().put("Content-Type", contentType);
```

Comment:
No need to change the response after it's built. In the original code just do:

    if (blob.getEncoding() != null) {
        type += "; charset=" + blob.getEncoding();
    }